elastic/integrations
# Add `cribl` integration package

## Proposed commit message

This PR adds the first phase of the new Cribl integration. 

This integration is unique in that it 1) has no input to run, since Cribl pushes data directly to elasticsearch via the _bulk API via their [elasticsearch output](https://docs.cribl.io/stream/destinations-elastic/), and 2) it doesn't really have its own pipeline, since the goal is to utilize existing integration's pipelines. It enables users to utilize the `reroute` processor to 'forward' logs from Cribl to a specific integration's datastream.

This first phase contains mostly documentation on how to set everything up using a custom pipeline. The second phase will contain a more elegant UI so that users can have a simpler experience configuring. 

## Checklist


## Author's Checklist



## How to test this PR locally



The steps to set everything up are documented in the integration's README

Then I just utilized the _bulk API (since that's what the cribl output uses) to verify the routing works

## Related issues


- Relates https://github.com/elastic/security-team/issues/7723

## Screenshots

<details>
  <summary>Integration main view</summary>
  
<img width="1695" alt="main" src="https://github.com/elastic/integrations/assets/9203126/1c384eb8-dcb5-4636-9ef8-d5df9fdc1b63">

</details>

<details>
  <summary>Instructions</summary>
  
<img width="697" alt="1" src="https://github.com/elastic/integrations/assets/9203126/f3f38552-4c18-42cc-9ee5-2125b9999cf5">

<img width="702" alt="2p1" src="https://github.com/elastic/integrations/assets/9203126/0d369649-ec22-4ced-aae5-be2c0605cc43">

<img width="690" alt="2p2" src="https://github.com/elastic/integrations/assets/9203126/56b33cf8-1bb4-49ab-88a2-d81e18d01bd3">


<img width="693" alt="3" src="https://github.com/elastic/integrations/assets/9203126/1ab52bbb-46ca-4f24-bcd3-dbab72b85865">


</details>
